data_analysis/plots.py

```diff
@@ -72,111 +288,464 @@ def boxplot(
+    kwargs['legend_kwds']['colorbar'] = False
+    lcm = LinearColormap(
+        colors=[
+            (13, 8, 135),
+            (84, 2, 163),
+            (139, 10, 165),
+            (185, 50, 137),
+            (219, 92, 104),
+            (244, 136, 73),
+            (254, 188, 43),
+            (240, 249, 33)
+        ],
+        vmin=0,
+        vmax=rider_gdf_geo[kwargs["column"]].max(),
+        caption=kwargs["legend_kwds"]["caption"]
+    )
+
+    lcm.add_to(plotted_map)
+
+    save_name = f'all_routes_numeric_{start_date}_to_{end_date}'
+
+    save_path = create_save_path(f'{save_name}_{kwargs["column"]}')
+    plotted_map.save(f"{save_path}.html")
+
+
+def groupby_long_df(df: pd.DataFrame,
+                    groupbyvars: Union[List[str], str]) -> pd.DataFrame:
+    """Group DataFrame by groupbyvars
+
+    Args:
+        df (pd.DataFrame): combined_long output DataFrame from
+            compare_scheduled_and_rt.main
+        groupbyvars Union[List[str], str]: A column or list
+            of columns to group by.
+
+    Returns:
+        pd.DataFrame: A DataFrame grouped by groupby
+            vars and a trip ratio column.
+    """
+    df = df.copy()
+    df['date'] = pd.to_datetime(df['date'])
 
+    df = (
+        df.groupby(groupbyvars)[['trip_count_rt', 'trip_count_sched']]
+        .sum()
+        .reset_index()
+    )
+
+    df['ratio'] = (
+        df['trip_count_rt']
+        / df['trip_count_sched']
+    )
+
+    return df
+
+
+def calculate_percentile_and_rank(df: pd.DataFrame) -> pd.DataFrame:
+    """Add a percentile and rank column to the DataFrame
+
+    Args:
+        df (pd.DataFrame): A summary DataFrame output from
+            compare_scheduled_and_rt.main
+
+    Returns:
+        pd.DataFrame: A summary DataFrame with the columns
+            percentiles and rank added.
+    """
+    df = df.copy()
+    df['percentiles'] = df['ratio'].rank(
+        pct=True
+    )
+    df['ranking'] = (
+        df['ratio']
+        .rank(method='dense', na_option='top', ascending=False)
+    )
+    return df
+
+
+def main(mvp: bool = False) -> None:
+    """Generate boxplot, maps of all routes, top 10 best routes, and
+    top 10 worst routes. Map of ridership
+
+    Args:
+        mvp (bool, optional): whether to generate plots and data
+            used in the MVP. Defaults to False.
+    """
+    logger.info("Creating GeoDataFrame")
+    gdf = static_gtfs_analysis.main()
+
+    logger.info("Getting latest real-time and schedule comparison data")
+
+    if mvp:
+        # Date range is between 2022-05-20 and 2022-11-05
+        logger.info(
+            "Loading data for MVP."
+            " Date range is between 2022-05-20 and 2022-11-05"
+        )
+        try:
+            combined_long_df = pd.read_csv(
+                DATA_PATH / 'combined_long_df_2022-11-06.csv'
+            )
+            summary_df = pd.read_csv(DATA_PATH / 'summary_df_2022-11-06.csv')
+        except FileNotFoundError:
```

Comment:
May I ask why we look in the parent path if we fail in the original directory?

Reply:
I originally had these saved in the child directory, which isn't under version control. I've committed those files to the parent directory now on the remote, so I can probably remove this `try-except` block.